handler/src/main/java/io/netty/handler/ssl/util/SelfSignedCertificate.java

```diff
@@ -332,7 +332,8 @@ public void delete() {
             wrappedBuf.release();
         }
 
-        fqdn = fqdn.replaceFirst("\\*", "X"); // Change asterisk to 'X'
+        // Change asterisk to 'x' for file name safety.
+        fqdn = fqdn.replaceFirst("[^\\w.-]", "x");
```

Comment:
why only `replaceFirst` and not `replaceAll` ?

Reply:
Well, wildcards are expected to have an asterisk at first char. Like this: `*.netty.io`. However, invalid fqdn such as `*.me*ow.io` is an invalid hostname. And `SelfSignedCertificate` must throw an exception.

Reply:
true... that said I think it may be better safe then sorry and so I would just use `replaceAll` and call it a day

Reply:
Alright, will change to `replaceAll`. :)